### PR TITLE
Standard Alfred scripts for pipeline

### PR DESCRIPTION
**Pull request by giorgiosironi**

### CODE REVIEW

**Comment by giorgiosironi** [comment]
As you can see Jenkins build this pull request, but to do so requires a `Jenkinsfile` committed in the project (think of it as the equivalent of `.travis.yml` but able to reuse our library of steps.)

**Comment by giorgiosironi** [comment]
@thewilkybarkid when you merge this, http://ci.alfred.elifesciences.org/job/library-api-sdk-php/ which is the build for the `master` branch will start to work.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 /composer.lock
 /vendor/
+/build/
```
Comment (thewilkybarkid) [inline .gitignore]: Could you alphabetise it?

Comment (giorgiosironi) [inline .gitignore]: Sure

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,4 @@
...
+*.swp
```
Comment (thewilkybarkid) [inline .gitignore]: Is this referring to Vim? If so, this should be the local `.gitignore` instead.

**Comment by thewilkybarkid** [comment]
Am I right in thinking there's only one build? For a library we'd want to recreate the various Travis environments (eg one for every supported PHP version, one for lowest dependency versions and possibly extra builds with certain dependency versions (eg if using Symfony using supported branches, eg 2.3, 2.7 and 2.8))). Or are you expecting a single build on Jenkins *plus* any number of custom builds on Travis?

**Comment by giorgiosironi** [comment]
At the moment only one build is performed, but I am looking into the best option for making multiple ones (either the `parallel` step or parameterized builds).

**Comment by giorgiosironi** [comment]
Added both current builds to `Jenkinsfile`. It's serial, not parallel, but it takes 10 seconds to run both:
http://ci.alfred.elifesciences.org/job/pull-requests/job/api-sdk-php/view/Pull%20Requests/job/PR-1/
rather than 1 minute and more on Travis:
https://travis-ci.org/elifesciences/api-sdk-php/builds/138874042
Is that enough?

File: phpunit.xml.dist [hunk 3]
```diff
@@ -15,5 +15,4 @@
             <directory>./src</directory>
         </whitelist>
     </filter>
-
```
Comment (thewilkybarkid) [inline phpunit.xml.dist]: Could you revert this?

**Comment by giorgiosironi** [comment]
Amended `phpunit.xml.dist`.

I have been checking out a docker container to build with PHP `7.1.0-alpha.1`, but it's missing all the extensions which are not trivial to build automatically from the C source.

**Comment by thewilkybarkid** [comment]
Ok, well maybe it'll be possible when it hits RC. I'm happy to merge and remove Travis, but the last commit seems to have failed.

**Comment by giorgiosironi** [comment]
Seems that rebasing a commit seriously broke it, and the relaunch appears green but may not be on the correct commit, investigating.

**Comment by giorgiosironi** [comment]
Cannot reproduce, I was confused by the fact that the commit that Jenkins builds is not `2f64a5a` (the top of this branch), but this new commit:
```
commit f7d042a5e71fb7ba621f2060eab0ebba149500e4
Merge: a80f7cf 2f64a5a
Author: Giorgio Sironi <info@giorgiosironi.com>
Date:   Tue Jun 21 16:27:04 2016 +0000

    Merge 2f64a5a073e9e9bef49d8b4118469c2a7b3490f0 into a80f7cf24999350ce5cee2a5eada144eafdd32f5

```
which is the result of merging this branch into `master`. The relaunched build correctly uses the most up-to-date and further rebases do not trigger errors.

**Comment by giorgiosironi** [comment]
Reporting the error at https://issues.jenkins-ci.org/browse/JENKINS-36137 so that it may be helpful for others.

**Pull request merged**